app.py

```diff
@@ -20,8 +40,10 @@
 # split data into chunks
 N_per_process = N // size
 
-# calculating kmeans in parallely
-data = X[rank * N_per_process : (rank + 1) * N_per_process]
+start_row = rank*N_per_process
+end_row = N_per_process*(rank+1)-1
+
+data = np.loadtxt(filename, delimiter=',', skiprows=start_row, max_rows=end_row-start_row+1)
```

Comment:
Something is smelly here, you've load data in line 31 and loading again here by doing a disk IO.
I'd rather 

1. Get a total row count in CSV. You can get the total count without reading the file as mentioned in here https://stackoverflow.com/a/16108605/6862650
2. Calculate N_per_process, determine from which line to which line each process should read
3. Pass the line range, use pd.read_csv to load csv. You can set starting lines and limit by parameters there